lib/engine-dylib/src/artifact.rs

```diff
@@ -812,7 +812,8 @@ impl Artifact for DylibArtifact {
 
         You need to be running MacOS for the following to actually work though.
         */
-        if path.extension().unwrap() == "dylib" {
+        let has_extension = path.extension().is_some();
+        if has_extension && path.extension().unwrap() == "dylib" {
```

Comment:
```suggestion
        if path.extension().unwrap_or_default() == "dylib" {
```

Reply:
This complained because it returns an OSString and not str. This seamed liked the cleanest solution?